chord_metadata_service/chord/ingest/phenopackets.py

```diff
@@ -138,62 +149,111 @@ def get_or_create_biosample(bs: dict) -> pm.Biosample:
+    gene_descriptor, _ = pm.GeneDescriptor.objects.get_or_create(
+        id=gene_desc["value_id"],
+        symbol=gene_desc["symbol"],
+        description=gene_desc.get("description", ""),
+        alternate_ids=gene_desc.get("alternate_ids", []),
+        xrefs=gene_desc.get("xrefs", []),
+        alternate_symbols=gene_desc.get("alternate_symbols", [])
+    )
+    return gene_descriptor
+
+
+def get_or_create_variant_descriptor(var_desc: dict) -> pm.VariationDescriptor:
+    gene_descriptor = _get_or_create_opt("gene_context", var_desc, get_or_create_gene_descriptor)
+    variant_descriptor, _ = pm.VariationDescriptor.objects.get_or_create(
+        id=var_desc["id"],
+        variation=var_desc.get("variation", {}),
+        label=var_desc.get("label", ""),
+        description=var_desc.get("description", ""),
+        gene_context=gene_descriptor,
+        expressions=var_desc.get("expressions", []),
+        vcf_record=var_desc.get("vcf_record", {}),
+        xrefs=var_desc.get("xrefs", []),
+        alternate_labels=var_desc.get("alternate_labels", []),
+        extensions=var_desc.get("extensions", []),
+        molecule_context=var_desc.get("molecule_context", "unspecified_molecule_context"),
+        structural_type=var_desc.get("structural_type", {}),
+        vrs_ref_allele_seq=var_desc.get("vrs_ref_allele_seq", ""),
+        allelic_state=var_desc.get("allelic_state", {})
+    )
+    return variant_descriptor
+
+
+def get_or_create_variant_interp(variant_interp_data: dict) -> pm.VariantInterpretation:
+    variant_descriptor = get_or_create_variant_descriptor(variant_interp_data["variation_descriptor"])
+    variant_interpretation, _ = pm.VariantInterpretation.objects.get_or_create(
+        acmg_pathogenicity_classification=variant_interp_data["acmg_pathogenicity_classification"],
+        therapeutic_actionability=variant_interp_data["therapeutic_actionability"],
+        variation_descriptor=variant_descriptor
+    )
+    return variant_interpretation
+
+
+def get_or_create_genomic_interpretation(gen_interp: dict) -> pm.GenomicInterpretation:
+    gene_descriptor = _get_or_create_opt("gene_descriptor", gen_interp, get_or_create_gene_descriptor)
+    variant_interpretation = _get_or_create_opt("variant_interpretation", gen_interp, get_or_create_variant_interp)
+
+    gen_obj, _ = pm.GenomicInterpretation.objects.get_or_create(
+        subject_or_biosample_id=gen_interp["subject_or_biosample_id"],
+        interpretation_status=gen_interp["interpretation_status"],
+        gene_descriptor=gene_descriptor,
+        variant_interpretation=variant_interpretation
     )
-    return g_obj
+    return gen_obj
 
 
 def get_or_create_disease(disease) -> pm.Disease:
     d_obj, _ = pm.Disease.objects.get_or_create(
         term=disease["term"],
         disease_stage=disease.get("disease_stage", []),
-        tnm_finding=disease.get("tnm_finding", []),
+        clinical_tnm_finding=disease.get("tnm_finding", []),
```

Comment:
shouldnt this prop access be `clinical_tnm_finding` for phenopackets v2 ?

Reply:
Good catch